### DISCUSSION TITLE
What's preventing the Storage SDK from being supported in Node?

### DISCUSSION THREAD

**Discussion by 0x6A75616E** [firebase/firebase-js-sdk#4682] [Q&A]
As part of a project, I'm building a node-based client that uploads files to Firebase storage. I found out that the firebase upload SDK is not supported in node.

I'd be able to upload to the bucket directly using the GCP lib, but that'd require a service key, which obviously I can't ship with my app. I need to use user authentication, which I've already implemented.

Is there a way around this? How would I be able to interact with firebase storage (or even GCP directly) from node using a user token?

**Comment by Feiyang1** [comment]
Thanks for the question! This has been a long standing feature request, but we haven't been able to prioritize it. Though I can't promise when we will add the support, we started talking about it internally recently.
Can you please follow https://github.com/firebase/firebase-js-sdk/issues/18 for progress?

**Comment by MarkoPetejan** [comment]
Add this to your node.js script:
global.XMLHttpRequest = require('xhr2');
global.WebSocket = require('ws');

**Comment by Feiyang1** [comment]
FYI, We are adding Nodejs support - https://github.com/firebase/firebase-js-sdk/pull/4985
